src/test/kotlin/sirius/db/jdbc/SmartQueryKotlinTest.kt

```diff
@@ -283,7 +283,10 @@ class SmartQueryKotlinTest {
         )
         val result = smartQueryTestChildEntity.queryList()
 
-        assertEquals(listOf("Parent 2"), result.stream().map { x -> x.name }.collect(Collectors.toList()))
+        assertEquals(
+            listOf("Parent 2", "Parent 3"),
+            result.stream().map { x -> x.name.substring(0, 8) }.collect(Collectors.toList())
```

Comment:
- Add a comment explaining this?
- Protect against `x.name.length` < 8?
- Compare sets (or sorted lists) if we cannot guarantee the order?

Reply:
I've added a comment about why we substring.
For the other comments: The data used in these test cases are inserted during the class setup, therefore controlled data, which means:
1. `x.name.length < 8` cannot happen
2. The order of items will be guaranteed by the ID (they are inserted in order). What we cannot guarantee is the oder in which tests are executed.